src/Axis3d.elm

```diff
@@ -250,6 +250,60 @@ intersectionWithPlane plane axis =
         Just (axisOrigin |> Point3d.translateIn axisDirection axialDistance)
 
 
+{-| Try to find the unique intersection point of an axis with a triangle. If
+the axis does not intersect the triangle, or if it is coplanar with it (lying
+perfectly in the plane of the triangle), returns `Nothing`.
+-}
+intersectionWithTriangle : Triangle3d units coordinates -> Axis3d units coordinates -> Maybe (Point3d units coordinates)
+intersectionWithTriangle (Types.Triangle3d ( p0, p1, p2 )) axis =
```

Comment:
Minor point but I try to avoid destructuring directly in argument lists, at least in published library code - it can mess with IDEs since often they'll show the parameter names during autocomplete or when hovering over the name of a function. So my preference would be to call this just `triangle` and then destructure inside the `let`.

Reply:
Good point! Fixed!